util/src/main/java/tech/pegasys/teku/util/config/TekuConfigurationBuilder.java

```diff
@@ -464,6 +483,8 @@ public TekuConfiguration build() {
           "eth1-deposit-contract-address is required if eth1-endpoint is specified.");
     }
 
+    // TODO validate config for validator node only
```

Comment:
Maybe here we should probably validate that, when running as a validator, the user isn't using some of the bacon node only options (and vice versa?).

Reply:
I think the validator only mode would need to be a subcommand (e.g. `teku validator_client` with alias `teku vc`) rather than a `--validator-only` kind of option.  Then users wouldn't be able to specify any options that weren't specifically allowed for the validator command.

Reply:
Do we need this TODO? And if so can it be replaced by an issue instead as that's more likely to actually be noticed...

Reply:
Yeah, we don't need it :)